edx_sga/static/js/src/edx_sga.js

```diff
@@ -314,10 +321,82 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                     });
                 block.find('#staff-debug-info-button')
                     .leanModal();
+
+                $(element).find('#download-init-button').click(function(e) {
+                  e.preventDefault();
+                  var self = this;
+                  $(self).addClass("disabled");
+                  $.get(
+                    prepareDownloadSubmissionsUrl,
+                    function(data) {
+                      if (data["downloadable"]) {
+                        window.location = downloadSubmissionsUrl;
+                        $(element).find('.task-message').hide();
+                      } else {
+                        $(element).find('.task-message').show();
+                        $(element).find('.task-message').html(comeBackMessage());
+                        $(element).find('.task-message').removeClass("ready-msg");
+                        $(element).find('.task-message').addClass("preparing-msg");
+                      }
+                      $(self).removeClass("disabled");
+                    }
+                  );
+                });
+                
+                pollUntilSuccess(downloadSubmissionsStatusUrl, checkResponse, 10000, 400).then(function(res) {
+                 if (res["zip_available"]) {
+                   $(element).find('.task-message').show();
+                   $(element).find('.btn-download-all').html(
+                     gettext("Download All Submissions (Ready)")
+                   );
+                   $(element).find('.task-message').html(gettext("Student submission file ready for download"));
+                   $(element).find('.task-message').removeClass("preparing-msg");
+                   $(element).find('.task-message').addClass("ready-msg");
+                 } else {
+                   $(element).find('.task-message').show();
+                   $(element).find('.task-message').html(comeBackMessage());
```

Comment:
As mentioned above, please use chaining for the `.task-message` element here and at line 352